gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -181,6 +184,34 @@ private ManagedChannel createSingleChannel() throws IOException {
     int port = Integer.parseInt(endpoint.substring(colon + 1));
     String serviceAddress = endpoint.substring(0, colon);
 
+    // TODO(hzyi): Use NettyChannelBuilder when maxHeaderListSize is specified to unblock Spanner.
+    // Changed to ManagedChannelBuilder when https://github.com/grpc/grpc-java/issues/4050 is resolved.
+    if (maxHeaderListSize != null) {
+      NettyChannelBuilder builder =
+          NettyChannelBuilder.forAddress(serviceAddress, port)
+              .intercept(headerInterceptor)
+              .intercept(metadataHandlerInterceptor)
+              .userAgent(headerInterceptor.getUserAgentHeader())
+              .executor(executor)
+              .maxHeaderListSize(maxHeaderListSize);
+      if (maxInboundMessageSize != null) {
+        builder.maxInboundMessageSize(maxInboundMessageSize);
+      }
+      if (keepAliveTime != null) {
+        builder.keepAliveTime(keepAliveTime.toMillis(), TimeUnit.MILLISECONDS);
+      }
+      if (keepAliveTimeout != null) {
+        builder.keepAliveTimeout(keepAliveTimeout.toMillis(), TimeUnit.MILLISECONDS);
+      }
+      if (keepAliveWithoutCalls != null) {
+        builder.keepAliveWithoutCalls(keepAliveWithoutCalls);
+      }
+      if (interceptorProvider != null) {
+        builder.intercept(interceptorProvider.getInterceptors());
+      }
+      return builder.build();
+    }
```

Comment:
I sometimes wish Java had macros.

Reply:
lol that was exactly what I was thinking when I was copying-pasting code for this PR.